core/sql/parser/sqlparser.y

```diff
@@ -13418,6 +13507,55 @@ query_specification : select_token '[' firstn_sorted NUMERIC_LITERAL_EXACT_NO_SC
           if (CmpCommon::getDefault(MVQR_LOG_QUERY_DESCRIPTORS) == DF_DUMP ||
               CmpCommon::getDefault(MVQR_LOG_QUERY_DESCRIPTORS) == DF_DUMP_MV)
             ((RelRoot*)$$)->setAnalyzeOnly();
+           } 
+/* type relx */
+query_specification : with_clause select_token '[' firstn_sorted NUMERIC_LITERAL_EXACT_NO_SCALE ']' set_quantifier query_spec_body
+	{
+	  if ($7) {
+	    $$ = new (PARSERHEAP())
+	      RelRoot(new (PARSERHEAP())
+		      GroupByAgg($8
```

Comment:
if($7) means there is a set_quantifier like (DISTINCT), so it will add a GroupByAgg node here.